Podcast/InternalProfileHeaderView.swift

```diff
@@ -0,0 +1,100 @@
+    
+    var viewProfileButton: UIButton!
+    var profileImage: UIImageView!
+    var nameLabel: UILabel!
+    var viewProfileLabel: UILabel!
+    var separator: UIView!
+    
+    var delegate: InternalProfileHeaderViewDelegate?
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        
+        backgroundColor = .podcastWhite
+        
+        profileImage = UIImageView(frame: .zero)
+        profileImage.layer.cornerRadius = profileImageSideLength / 2
+        profileImage.clipsToBounds = true
+        profileImage.backgroundColor = .podcastWhiteDark
+        addSubview(profileImage)
+        
+        nameLabel = UILabel(frame: .zero)
+        nameLabel.font = .systemFont(ofSize: 14, weight: UIFontWeightSemibold)
+        nameLabel.textColor = .podcastBlack
+        nameLabel.textAlignment = .left
+        addSubview(nameLabel)
+        
+        viewProfileLabel = UILabel(frame: .zero)
+        viewProfileLabel.font = .systemFont(ofSize: 12, weight: UIFontWeightRegular)
+        viewProfileLabel.textColor = .podcastTealBackground
+        viewProfileLabel.textAlignment = .left
+        viewProfileLabel.text = "View your profile"
+        addSubview(viewProfileLabel)
+        
+        separator = UIView(frame: .zero)
+        separator.backgroundColor = .podcastGray
+        addSubview(separator)
+        
+        viewProfileButton = UIButton(type: .custom)
+        viewProfileButton.backgroundColor = .clear
+        viewProfileButton.addTarget(self, action: #selector(viewProfilePressed), for: .touchUpInside)
+        addSubview(viewProfileButton)
+        
+    }
+    
+    override func layoutSubviews() {
+        profileImage.frame = CGRect(x: profileImageX, y: (height - profileImageSideLength) / 2, width: profileImageSideLength, height: profileImageSideLength)
+        profileImage.layer.cornerRadius = profileImageSideLength / 2
+        nameLabel.frame = CGRect(x: profileImageX + profileImageSideLength + padding, y: nameLabelY, width: frame.width - (profileImageX + profileImageSideLength + 2 * padding), height: nameLabelHeight)
+        viewProfileLabel.frame = CGRect(x: profileImageX + profileImageSideLength + padding, y: viewProfileLabelY, width: frame.width - (profileImageX + profileImageSideLength + 2 * padding), height: viewProfileLabelHeight)
+        viewProfileButton.frame = CGRect(x: 0, y: 0, width: frame.width, height: frame.height)
+        separator.frame = CGRect(x: 0, y: frame.height - separatorHeight, width: frame.width, height: separatorHeight)
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    func viewProfilePressed() {
+        delegate?.internalProfileHeaderViewDidPressViewProfile(internalProfileHeaderView: self)
+    }
+    
+    func setUser(_ user: User) {
+        nameLabel.text = user.name
+        if let url = user.imageURL {
+            if let data = try? Data(contentsOf: url) {
```

Comment:
bad this will load synchronously, okay for now but we will need to replace this later